pytheus/registry.py

```diff
@@ -0,0 +1,63 @@
+from threading import Lock
+from typing import Iterable, Protocol
+
+
+class Collector(Protocol):
+    name: str
+
+    def collect(self) -> Iterable:
+        pass
+
+
+class Registry(Protocol):
```

Comment:
A `Registry` is a collector so it would make sense for this to be a `Registry(Collector, Protocol):` without repeating the collect requirement.

But I can see that we have a `name` attribute on `Collector`, following the prometheus docs to the line we could remove it and treat a collector as anything with a `collect()` method, but I can also see our implementation requirements so I'm not completely sure 🤔

Reply:
The problem with removing the property, mypy would complain `"Collector" has no attribute "name"`

Reply:
this possibly requires more analysis, but definitely not now!